src/main/java/com/gtnewhorizons/gtnhgradle/modules/JVMDowngraderModule.java

```diff
@@ -0,0 +1,595 @@
+                    constraint -> constraint.because("Required for JVM Downgrader stubs when not shading them"));
+        }
+    }
+
+    private void ensureJvmdgSnapshotRepo(Project project) {
+        if (UpdateableConstants.JVMDG_VERSION.contains("-SNAPSHOT")) {
+            project.getRepositories()
+                .maven(repo -> {
+                    repo.setName("WagYourTail Maven Snapshots");
+                    repo.setUrl(UpdateableConstants.JVMDG_SNAPSHOT_REPO);
+                    repo.mavenContent(content -> content.snapshotsOnly());
+                    repo.content(desc -> desc.includeGroup(UpdateableConstants.JVMDG_GROUP));
+                });
+        }
+    }
+
+    private Configuration createDowngradedApiConfiguration(Project project, ConfigurationContainer cfgs,
+        int downgradeTarget) {
+        final String configName = "jvmdgApiForJava" + downgradeTarget;
+        return cfgs.create(configName, config -> {
+            config.setDescription("JVM Downgrader API jar for Java " + downgradeTarget);
+            config.setCanBeConsumed(false);
+            config.setCanBeResolved(true);
+            config.setVisible(false);
+            config.setTransitive(false);
+            config.getDependencies()
+                .add(
+                    project.getDependencies()
+                        .create(UpdateableConstants.jvmdgApiDependency(downgradeTarget)));
+        });
+    }
+
+    /** Adds Jabel stub jar for @Desugar annotation compatibility. */
+    private void addJabelStub(Project project, DependencyHandler deps) {
+        final TaskProvider<?> extractStub = project.getTasks()
+            .register("extractJabelStub", task -> {
+                final var outputFile = project.getObjects()
+                    .fileProperty();
+                outputFile.set(
+                    project.getLayout()
+                        .getBuildDirectory()
+                        .file("gtnhgradle/jabel-stub/jabel-stub.jar"));
+                task.getOutputs()
+                    .file(outputFile);
+                task.doLast(t -> {
+                    try (InputStream stubStream = getClass().getResourceAsStream("/jabel-stub/jabel-stub.jar")) {
+                        if (stubStream == null) {
+                            throw new GradleException("Jabel stub jar not found in GTNHGradle resources");
+                        }
+                        final File stubJar = outputFile.get()
+                            .getAsFile();
+                        stubJar.getParentFile()
+                            .mkdirs();
+                        Files.copy(stubStream, stubJar.toPath(), StandardCopyOption.REPLACE_EXISTING);
+                    } catch (IOException e) {
+                        throw new GradleException("Failed to extract Jabel stub jar", e);
+                    }
+                });
+            });
+
+        deps.add(JavaPlugin.COMPILE_ONLY_CONFIGURATION_NAME, project.files(extractStub));
```

Comment:
We just need to publish this as a regular maven dep, this is a weird thing for a plugin to do outside of complying with Minecraft's EULA etc.